php/libraries/NDB_Client.class.inc

```diff
@@ -89,7 +89,7 @@ class NDB_Client
         if (!$this->_isWebClient) {
             // Set user as unix username.
             // Requires Loris to have user with this username
-            $user =& User::singleton(get_current_user());
+            $user =& User::singleton(getenv('USER'));
```

Comment:
I don't think `$user` is ever used. There is also an other affectation at line #176. They should be removed.
If the only purpose of this is to crash when it fail, then a Exception/error message should be thrown.

Unless I am missing something...

Reply:
I updated the pull request description. Hopefully it makes sense. It's a somewhat simple, yet deep fix for a wrong exception that was thrown in the fix_timepoint_date_problems.php script. It's only supposed to crash when you try to run a script with a corresponding username that doesn't exist in LORIS(I'm sure there's a reason for this design...). Otherwise it shouldn't falsely crash when you're doing everything right.

You're right about `$user` not being used.

Reply:
getenv("USER") is better than get_current_user() for the intended purpose. Still, we should remove those lines and use lazy instanciation instead of this "effet de bors"

Reply:
@xlecours pretty sure thats to allow the DB to trackchanges when you run backend scripts in the history tables. it uses the unix user as lorisuser.

its annoying because it requires you to have a loris user with the same userid as ur unix user. and when it comes to
get_current_user it was actually checking the owner of the PHP file rather than the actual logged in unix user..

I'm not 100% on it but @driusan should be able to tell if its true or not.